### DISCUSSION TITLE
Instructions to create a local story-ramp package for testing

### DISCUSSION THREAD

**Discussion by RyanCoulsonCA** [ramp4-pcar4/story-ramp#398] [Announcements]
From @yileifeng, ~~for some reason he can't view/post discussions~~ fixed now that I've created this one 🏆 .

Here is a method to test any changes made to the storylines package in the editor locally (mainly for work related to preview feature):

1. In story-ramp repo, make required changes
2. `npm run build-plugin` - to build as plugin
3. `npm pack` - produces file called ramp-storylines_demo-scenarios-pcar-[version].tgz
4. Navigate to editor repo
5. Change ramp-storylines_demo-scenarios-pcar dependency in package.json to point towards the packed file (e.g. "ramp-storylines_demo-scenarios-pcar": "../storylines/ramp-storylines_demo-scenarios-pcar-3.0.4.tgz")
6. `npm install` and run editor

**Comment by yileifeng** [comment]
Method to publish to NPM package https://www.npmjs.com/package/ramp-storylines_demo-scenarios-pcar:

1. Update version number in `package.json`
2. `npm run build-plugin` - to build as plugin
3. `npm login`
4. `npm publish`
